hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -282,11 +275,12 @@ protected void processNextDeletedRecord(DeleteRecord deleteRecord) {
 
       Comparable curOrderingVal = oldRecord.getOrderingValue(this.readerSchema, this.hoodieTableMetaClient.getTableConfig().getProps());
       Comparable deleteOrderingVal = deleteRecord.getOrderingValue();
+      Pair<Comparable, Comparable> comparablePair = OrderingValueUtils.canonicalizeOrderingValue(curOrderingVal, deleteOrderingVal);
       // Checks the ordering value does not equal to 0
       // because we use 0 as the default value which means natural order
       boolean choosePrev = !deleteOrderingVal.equals(0)
-          && ReflectionUtils.isSameClass(curOrderingVal, deleteOrderingVal)
-          && curOrderingVal.compareTo(deleteOrderingVal) > 0;
+              && ReflectionUtils.isSameClass(comparablePair.getLeft(), comparablePair.getRight())
```

Comment:
we can reassign `curOrderingVal` and `deleteOrderingVal` with canonicalized value, to improve the  readability of code.